TabloidMVC/Repositories/PostRepository.cs

```diff
@@ -91,6 +93,50 @@ public Post GetPublishedPostById(int id)
             }
         }
 
+        //This creates a list of all posts published by the user who is currently logged in.
+        public List<Post> GetAllUsersPosts(int userProfileId)
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                       SELECT p.Id, p.Title, p.Content, 
+                              p.ImageLocation AS HeaderImage,
```

Comment:
This is a minor note, but depending on the size of your SQL query it can really slow down your application-- rather than getting EVERYTHING from all the tables, try only getting what you need to display! So in this case it's just title, author, and category (plus anything else you need for JOINs or WHEREs). The less unnecessary data you pull from your DB, the faster your code will run!

Reply:
Thank you! With permission, leaving as is until a possible later date once I get all Posts functionality tickets completed.

Note on that: will likely need to create something similar to Add(NewPostFromReader) method designed for GetAllUsersPosts to align with the pared down SQL query.